Userland/Applications/Spreadsheet/Spreadsheet.h

```diff
@@ -25,15 +25,20 @@ namespace Spreadsheet {
 class CellChange {
 public:
     CellChange(Cell&, DeprecatedString const&);
+    CellChange(Cell&, CellTypeMetadata const&);
 
     auto& cell() { return m_cell; }
     auto& previous_data() { return m_previous_data; }
     auto& new_data() { return m_new_data; }
+    auto& previous_type_metadata() { return m_previous_type_metadata; }
+    auto& new_type_metadata() { return m_new_type_metadata; }
 
 private:
     Cell& m_cell;
     DeprecatedString m_previous_data;
     DeprecatedString m_new_data;
+    CellTypeMetadata m_previous_type_metadata;
+    CellTypeMetadata m_new_type_metadata;
```

Comment:
Some of these fields are uninhabited with your change, I'd rather have only the available fields active (i.e. in a variant, or a union given that the discriminator is not actually within this class).

Reply:
I'm not sure how to do this.

Would it be something like:
```cpp
struct Data {
	DeprecatedString previous_data;
	DeprecatedString new_data;
} Data;

struct TypeMetadata {
	CellTypeMetadata previous_type_metadata;
	CellTypeMetadata new_type_metadata;
} TypeMetadata;

Variant<Data, TypeMetadata> m_active_data;
```

Reply:
I guess that could work, yeah.
this one isn't that important, you can ignore the comment if you want.

Reply:
This one is making my head spin so I'm going to ignore for now and I'll open a new PR if I figure it out.